falcon/cluster/cluster.py

```diff
@@ -272,191 +358,97 @@ def _dist_mz_interval(
     indptr_i : int
         The current start index in `indptr`.
     """
-    for batch_start in range(0, vectors.shape[0], batch_size):
-        batch_stop = min(batch_start + batch_size, vectors.shape[0])
+    batch_start = 0
+    nn_mz = _get_mz_neighbors(
+        precursor_mzs, precursor_tol_mass, precursor_tol_mode
+    )
+    if rt_tol is not None:  # TODO: test this
+        nn_rt = _get_mz_neighbors(rts, rt_tol, "rt")
+        nn_mz = [np.intersect1d(mz, rt) for mz, rt in zip(nn_mz, nn_rt)]
+    for batch in dataset.to_batches(batch_size=batch_size, filter=query):
+        batch_spectra = (
+            batch.to_pandas().apply(spectrum.df_row_to_spec, axis=1).tolist()
+        )
+        batch_spectra = nb.typed.List(batch_spectra)
+        vectors = vectorize(spectra=batch_spectra)
```

Comment:
I was thinking about vectorizing the spectra when they are processed and adding the vectors to the lance dataset, but I don't think this will speed up the whole process. The current implementation is quite efficient.

Reply:
I tried it and it's about 1 sec faster when clustering 70k spectra